Style Guide and Diataxis Review: Hornet II (#1322)
gohornet/hornet: documentation/docs/post_installation/peering.md

## Node Identity

Each node can be uniquely identified by a `peer identity`. `Peer identity` (also called `PeerId`) is represented by a public
and private key pair. Since `PeerId` is a cryptographic hash of a peer's public key, the `PeerId` represents a verifiable link between the given peer and its public key. It enables individual peers to establish a secure communication channel as the hash can be used to verify an identity of the peer.

> When Hornet is started for the first time, it will automatically generate a `PeerId` and save the identity's private key in the `./p2pstore/identity.key` file. Hornet will keep the generated identity between subsequent restarts.

Reviewer: ```suggestion
When you start Hornet for the first time, it will automatically generate a `PeerId` and save the identity's private key in the `./p2pstore/identity.key` file. Hornet will keep the generated identity between subsequent restarts.
```

/ip4/100.1.1.1/tcp/15600/p2p/12D3KooWHjcCgWPnUEP8wNdbL2fx63Cmosk16xyZ25iUZagxmHb4
```

:::note

> Consider how `ip4` is used. A common mistake is to use `ipv4`.

Reviewer: Is "consider" the right word here? It sounds a bit weird

/dns/node01.iota.org/tcp/15600/p2p/12D3KooWHjcCgWPnUEP8wNdbL2fx63Cmosk16xyZ25iUZagxmHb4
```

You will need to find out your own `multiaddr` to give to your peers for neighboring. To do so, combine the `peerId` you received
from the stdout when the Hornet node started up (or which was shown via the `p2pidentity-gen` CLI tool), and your
configured `p2p.bindAddress`. Replace the `/ip4/<ip_address>`/`/dns/<hostname>` segments with the actual information.

You can find more information about `multiaddr` at the [libp2p docs page](https://docs.libp2p.io/concepts/addressing/).

## Adding Node Peers

> Once you know your node's own `multiaddr`, it can be exchanged with other node owners to establish a mutual peer connection. We recommended several peer neighbors between 4-6 to get some degree of redundancy.

Reviewer: ```suggestion
Once you know your node's own `multiaddr`, it can be exchanged with other node owners to establish a mutual peer connection. We recommended between 4 to 6 peer neighbors to get some degree of redundancy.
```

## Finding Neighbors

You can join the official IOTA Discord server and the `#nodesharing` channel. There, you will be able to describe your node location (Europe, 
Germany, Asia, etc.), with your allocated high watermark resources and ask for neighbors.

> :::note

Reviewer: This feels like it needs more severity than a note.

Do not publicly disclose your node `multiaddr` to all readers but wait for an individual direct chat.

:::

> You can add peers using the Hornet [dashboard](https://wiki.iota.org/hornet/post_installation#dashboard). To do so, go to *Peers* and click on *Add Peer*. You can also add peers on the [peering.json](peering.md) file.

Reviewer: ```suggestion
You can add peers using the [Hornet dashboard](https://wiki.iota.org/hornet/post_installation#dashboard). To do so, go to *Peers* and click on *Add Peer*. You can also add peers on the [peering.json](peering.md) file.
```

## Autopeering

Hornet also supports automatically finding peers through the _autopeering_ module. To minimize service distribution in case your autopeered peers are flaky, we recommend you only use autopeering if you have at least four static peers.

Autopeering is disabled by default. If you want to enable it, add `"Autopeering"` to `node.enablePlugins`.

:::note 

The autopeering plugin will disclose your public IP address to possibly all nodes and entry points.
Do not enable this plugin if you do not want this to happen!

:::

Your node will use the specified entry nodes under `p2p.autopeering.entryNodes` to find new peers. `entryNodes` are also encoded as `multiaddr`:

```
/ip4/45.12.34.43/udp/14626/autopeering/8CZELJwB3aBzxJgnLMvvt1FirAwNN6jif9LavYTNHCty
```

> where the `/autopeering` portion defines the base58 encoded Ed25519 public key.

Reviewer: ```suggestion
* Where the `/autopeering` portion defines the base58 encoded Ed25519 public key.
```

By default, Hornet will peer up to four autopeered peers and initiate a gossip protocol with them. Autopeered peers are not subject to connection trimming, the same way as mutually tethered peers are not either.

### Entry Node

If you want to run your own node as an autopeering entry node, you should enable `p2p.autopeering.runAsEntryNode`. The base58 encoded public key is in the output of the `p2pidentity-gen` Hornet tool. Alternatively, if you already have an identity in a `./p2pstore`, you can use the `p2pidentity-extract` Hornet tool to extract it.

### Low/High Watermark

> The `p2p.connectionManager.highWatermark` and `p2p.connectionManager.lowWatermark` configuration options define "watermark" points. Watermark points can be considered like a filling basin where if the `highWatermark` is reached, water will be drained until it reaches the `lowWatermark` again. Similarly, the connection manager within Hornet will start trimming away connections to peers if `highWatermark` peers are connected until it reaches `lowWatermark` count of peers. These watermarks exist for a certain buffer number of peers to be connected, which will not necessarily be targeted by the gossip protocol.

Reviewer: ```suggestion
The `p2p.connectionManager.highWatermark` and `p2p.connectionManager.lowWatermark` configuration options define "watermark" points. You can think of watermark points like a filling basin where if the `highWatermark` is reached, water will be drained until it reaches the `lowWatermark` again. Similarly, the connection manager within Hornet will start trimming away connections to peers if `highWatermark` peers are connected until it reaches `lowWatermark` count of peers. These watermarks exist for a certain buffer number of peers to be connected, which will not necessarily be targeted by the gossip protocol.
```
